### ISSUE TITLE
Subtasks with Reminders app for macOS Catalina

### ISSUE THREAD

**Issue by devurandom** [etesync/etesync-dav#127]
The preinstalled "Reminders" app on macOS Catalina 10.15.3 does not show the subtasks of my task list. Tasks.org on Android displays them fine.

The setup is:
* Primary account
  - Android
    - Tasks.org > EteSync
* Associate account
  - macOS
    - System Settings > Internet Accounts > CalDAV
    - Reminders

**Comment by tasn** [comment]
Could you please show an example here from the journal? Feel free to censor the summary line, but make sure to include both the task, and the subtask (that's not showing). I wonder if it's a problem with etesync-dav or if it's a limitation of the macOS reminders app.

**Comment by devurandom** [comment]
Child:
```ical
BEGIN:VCALENDAR
VERSION:2.0
PRODID:+//IDN tasks.org//android-746//EN
BEGIN:VTODO
DTSTAMP:20200501T230556Z
UID:2135877053792992860
SEQUENCE:1
CREATED:20200417T170217Z
LAST-MODIFIED:20200501T164934Z
SUMMARY:REDACTED
STATUS:NEEDS-ACTION
RELATED-TO;RELTYPE=PARENT:276800902850960496
END:VTODO
END:VCALENDAR
```

Parent:
```ical
BEGIN:VCALENDAR
VERSION:2.0
PRODID:+//IDN tasks.org//android-746//EN
BEGIN:VTODO
DTSTAMP:20200501T230557Z
UID:276800902850960496
SEQUENCE:1
CREATED:20200328T141014Z
LAST-MODIFIED:20200501T230428Z
SUMMARY:REDACTED
STATUS:NEEDS-ACTION
CATEGORIES:Shopping
END:VTODO
END:VCALENDAR
```

They display fine (as parent and child) in Tasks.org.

**Comment by devurandom** [comment]
> include both the task, and the subtask (that's not showing)

Sorry, I miscommunicated that part: The subtasks are showing, but not as subtasks, but as top-level tasks instead.

**Comment by tasn** [comment]
Looking at the VTODO it looks correct. I have two more things to check if you are able to:
1. Open the etesync-dav URL for this task list in the browser and download the `.ics`. Do these two tasks indeed show like this?
2. Create a new task and a new subtask from the macOS reminders app. I wonder how those will look like. Because if macOS doesn't understand this (standard compliant!) way of defining the relationship, I wonder what format they use. Maybe we can workaround their oddities in our code.

**Comment by devurandom** [comment]
> 1. Open the etesync-dav URL for this task list in the browser and download the `.ics`. Do these two tasks indeed show like this?

This is what I see when I look at the ICS file on my own Linux computer (assuming that etesync-dav outputs the same data on Linux and macOS):
```ics
BEGIN:VTODO
CATEGORIES:Shopping
CREATED:20200328T141014Z
DTSTAMP:20200501T230557Z
LAST-MODIFIED:20200501T230428Z
SEQUENCE:1
STATUS:NEEDS-ACTION
SUMMARY:REDACTED
UID:276800902850960496
END:VTODO
```

```ics
BEGIN:VTODO
CREATED:20200417T170217Z
DTSTAMP:20200501T230556Z
LAST-MODIFIED:20200501T164934Z
RELATED-TO;RELTYPE=PARENT:276800902850960496
SEQUENCE:1
STATUS:NEEDS-ACTION
SUMMARY:REDACTED
UID:2135877053792992860
END:VTODO
```

I will later have to check and confirm that the output is indeed the same on macOS.

> 2. Create a new task and a new subtask from the macOS reminders app. I wonder how those will look like. Because if macOS doesn't understand this (standard compliant!) way of defining the relationship, I wonder what format they use. Maybe we can workaround their oddities in our code.

This will have to wait a bit, as I will have to coordinate with the Mac's owner.

**Comment by tasn** [comment]
> This is what I see when I look at the ICS file on my own Linux computer (assuming that etesync-dav outputs the same data on Linux and macOS):

The ICS looks correct, and it's indeed the same across platforms, so no need to verify on the mac.

> This will have to wait a bit, as I will have to coordinate with the Mac's owner.

Please do and let me know. They just need to create a task and a subtask and then you can check yourself what shows up in the journal and let me know here. It's a simple as that.


Thanks a lot!

**Comment by devurandom** [comment]
Off-Topic: I got to play for a few minutes with the iOS Reminders app. According to the docs [1] it should be possible to drag & drop a task onto another task in order to make the former a subtask of the latter. This did not work with in the list in the EteSync (fake) CalDAV account and it did not work with a list in the "local" account. Only after I enabled iCloud sync of Reminders and the "local" account turned into an "iCloud" account in the Reminders app, I was able to create subtasks in the manner described in the docs in task lists of that account. After I disabled iCloud sync of Reminders creating subtasks stopped working. To emphasize: This was on iOS and not macOS. But maybe the Reminders app on both follows a similar logic.

[1]: https://support.apple.com/en-us/HT205890

**Comment by tasn** [comment]
Thanks for the update. It sounds quite plausible (and consistent with how Apple does things) that subtasks are just not supported for non-iCloud accounts.

I just checked [the EventKit API docs](https://developer.apple.com/documentation/eventkit/creating_events_and_reminders) again, and there indeed doesn't seem to be an easy way. I have an EteSync backed Reminders app for iOS I built a few months ago. It's mostly similar to Tasks.org, and is quite polished, just not well tested. Maybe it's time to take it to the next level.
